tests/src/gcc/LaunchKernel.c

```diff
@@ -0,0 +1,180 @@
+
+  if(A == 333)
+    printf("Test Passed!!\n\n");
+  else
+    printf("Test Failed :Value of A is %d\n\n",A);
+
+  hipFree(A_d);
+}
+
+void LaunchKernelArg2()
+{
+  hipError_t status = hipSuccess;
+  int A = 0;
+  int B = 123;
+  int *A_d = NULL;
+  int *B_d = NULL;
+
+  dim3 blocks       = {1,1,1};
+  dim3 threads      = {1,1,1};
+
+  printf(" -----: Launch Two Argument Kernel :------ \n");
+
+  // Allocate Device memory
+  status = hipMalloc((void**)&A_d, sizeof(int));
+  printf("hipMalloc status %s\n", hipGetErrorString(status));
+
+  status = hipMalloc((void**)&B_d, sizeof(int));
+  printf("hipMalloc status %s\n", hipGetErrorString(status));
+
+  // Copy data from host memory to device memory
+  status = hipMemcpy(B_d,&B, sizeof(int), hipMemcpyHostToDevice);
+  printf("hipMalloc status %s\n", hipGetErrorString(status));
+
+
+  void* Args[]={A_d,B_d};
+  hipLaunchKernel(kernel2, blocks, threads, Args,0,0);
+  printf("hipLaunchKernel status %s\n", hipGetErrorString(status));
+
+  // Get the result back to host memory
+  status = hipMemcpy(&A, A_d, sizeof(int), hipMemcpyDeviceToHost);
+  printf("hipMalloc status %s\n", hipGetErrorString(status));
+
+  if(A == 123)
+    printf("Test Passed!! \n\n");
+  else
+    printf("Test Failed : Value of A is %d\n\n",A);
+
+  hipFree(A_d);
+  hipFree(B_d);
+}
+
+void LaunchKernelArg3()
+{
+  hipError_t status = hipSuccess;
+  int A = 321;
+  int B = 123;
+  int C = 0;
+  int *A_d = NULL;
+  int *B_d = NULL;
+  int *C_d = NULL;
+
+  dim3 blocks       = {1,1,1};
+  dim3 threads      = {1,1,1};
+
+  printf(" -----: Launch Three Argument Kernel :------ \n");
+
+  // Allocate Device memory
+  status = hipMalloc((void**)&A_d, sizeof(int));
+  printf("hipMalloc status %s\n", hipGetErrorString(status));
+
+  status = hipMalloc((void**)&B_d, sizeof(int));
+  printf("hipMalloc status %s\n", hipGetErrorString(status));
+
+  status = hipMalloc((void**)&C_d, sizeof(int));
+  printf("hipMalloc status %s\n", hipGetErrorString(status));
+
+  // Copy data from host memory to device memory
+  status = hipMemcpy(A_d,&A, sizeof(int), hipMemcpyHostToDevice);
+  printf("hipMalloc status %s\n", hipGetErrorString(status));
+
+  status = hipMemcpy(B_d,&B, sizeof(int), hipMemcpyHostToDevice);
+  printf("hipMalloc status %s\n", hipGetErrorString(status));
+
+
+  void* Args[]={A_d,B_d,C_d};
+  hipLaunchKernel(kernel3, blocks, threads, Args,0,0);
+  printf("hipLaunchKernel status %s\n", hipGetErrorString(status));
+
+  // Get the result back to host memory
+  status = hipMemcpy(&C, C_d, sizeof(int), hipMemcpyDeviceToHost);
+  printf("hipMalloc status %s\n", hipGetErrorString(status));
+
+  if(C == 444)
+    printf("Test Passed\n\n");
+  else
+    printf("Test Failed : Value of C is %d\n\n",C);
+
+  hipFree(A_d);
+  hipFree(B_d);
+  hipFree(C_d);
+}
+
+
+int main()
+{
+  LaunchKernelArg();
+  LaunchKernelArg1();
+  LaunchKernelArg2();
+  LaunchKernelArg3();
+
+  printf("PASSED!\n");
```

Comment:
Will this test always PASS?

Reply:
The right way to do it is that each of the LaunchKernelArg* functions returns a bool based on pass or fail. You declare a bool in main whose initial value is true. And you AND the return value from each of these LaunchKernelArg functions. Then you check the final return value to determine PASS or FAIL.